testsuite/MDAnalysisTests/analysis/test_hbonds.py

```diff
@@ -397,3 +397,81 @@ def test_table_atoms(self, h, normalized_timeseries, reference_table):
         # https://github.com/MDAnalysis/mdanalysis/issues/801)
         for name, ref in reference_table.items():
             assert_array_equal(h.table.field(name), ref, err_msg="resname for {0} do not match (Issue #801)")
+
+
+class TestHydrogenBondAnalysisTIP3PHeavyPBC(object):
+    @staticmethod
+    @pytest.fixture()
+    def universe():
+        return MDAnalysis.Universe(waterPSF, waterDCD)
+
+    kwargs = {
+        'selection1': 'all',
+        'selection2': 'all',
+        'detect_hydrogens': "distance",
+        'distance': 3.0,
+        'angle': 120.0,
+        'distance_type': 'heavy',
+        'pbc': True
+    }
+
+    @pytest.fixture()
+    def h(self, universe):
+        h = MDAnalysis.analysis.hbonds.HydrogenBondAnalysis(universe, **self.kwargs)
+        h.run(verbose=False)
+        h.generate_table()
+        return h
+
+    @pytest.fixture()
+    def normalized_timeseries(self, h):
+        # timeseries in normalized form: (t, d_indx1, a_indx1, d_index0, a_index0, donor, acceptor, dist, angle)
+        #                   array index:  0     1        2        3         4        5      6        7      8
+        timeseries = [[t] + item
+                      for t, hframe in zip(h.timesteps, h.timeseries)
+                      for item in hframe]
+        return timeseries
+
+    # keys are the names in the h.table
+    reference = {
+        'distance': {'mean': 2.7627309, 'std': 0.0905052},
+        'angle': {'mean': 158.9038039, 'std': 12.0362826},
+    }
+
+    @pytest.fixture()
```

Comment:
add scope="class" so that it only gets run once – works fine if data are only read.

Reply:
I've added this.